src/pages/index.js

```diff
@@ -43,73 +106,189 @@ const initialCards = [
-    cardElement.remove();
+    handleDeleteCard(cardElement, cardData);
   });
 
-  cardImageElement.addEventListener("click", () => {
-    openModal(previewModal);
+  // Set initial like state
+  if (cardData.isLiked) {
+    cardLikeButton.classList.add("card__like-btn_liked");
+  }
 
-    previewModalImageElement.src = data.link;
-    previewModalImageElement.alt = data.name;
-    previewModalCaptionElement.textContent = data.name;
+  // Open the card preview
+  cardImage.addEventListener("click", () => {
+    previewModalImageElement.src = cardData.link;
+    previewModalImageElement.alt = cardData.name;
+    previewModalCaptionElement.textContent = cardData.name;
+    openModal(previewModal);
   });
 
+  // Add click event listener for like button
+  cardLikeButton.addEventListener("click", () => {
+    handleLikeClick(cardData._id, cardLikeButton);
+  });
   return cardElement;
 }
 
+// Open Avatar Modal
+profileAvatarBtn.addEventListener("click", () => {
+  resetValidation(avatarForm, settings);
+  openModal(avatarModal);
+});
+
+// Close Avatar Modal
+avatarModalCloseBtn.addEventListener("click", () => closeModal(avatarModal));
+
+// Avatar Form Submission
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
 function openModal(modal) {
   modal.classList.add("modal_opened");
   document.addEventListener("keydown", handleEscapeKey);
 }
+// Handle avatar update submission
+// Avatar submission handler
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = avatarForm.querySelector(".modal__submit-btn");
+  const avatarImg = document.querySelector(".profile__avatar");
+
+  setSubmitButtonState(submitButton, true, "Saving...", "Save");
+
+  api
+    .updateAvatar({ avatar: avatarUrlInput.value })
+    .then((userData) => {
+      avatarImg.src = `${userData.avatar}?t=${Date.now()}`;
+      avatarForm.reset();
+      closeModal(avatarModal);
+      resetValidation(avatarForm, settings);
+      // Disable the button after successful submission
+      disableButton(submitButton, settings);
+    })
+    .catch((err) => {
+      console.error("Avatar update failed:", err);
+      alert("Failed to update avatar. Please try again.");
+    })
+    .finally(() => {
+      setSubmitButtonState(submitButton, false, "Saving...", "Save");
+      // Make sure the button stays disabled even after the loading state is removed
+      disableButton(submitButton, settings);
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = cardForm.querySelector(".modal__submit-btn");
+
+  setSubmitButtonState(submitButton, true, "Creating...", "Create");
+
+  api
+    .addCard({
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    })
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      cardForm.reset();
+      closeModal(cardModal);
+      resetValidation(cardForm, settings);
```

Comment:
`resetValidation` is not needed after **submission** because there can't be any validation errors if the submission is successful